### ISSUE TITLE
Support for Gnome 46

### ISSUE THREAD

**Issue by negesti** [negesti/gnome-shell-extensions-negesti#208]
After skipping Fedora 39 I upgrade to 40 last week and upgrade the extension. 

- [ ]  remove Application based config (see #207 )
  - no need for wnck any more 
   #191 , 
   #184 ,
   #172, 
   #150, 
   #138,
    #142,
    #134,
    #118,
 #108
- [ ] It is untested for v45.0 #206
- [ ] Wayland finally support rdp and screen sharing, so all the old wnck an
  - The extension should work with X11 (atm)
- [ ] test / fix MoveFocus for v46.0

**Comment by TheRoarkster** [comment]
Welcome back!  Great to have this under development again.  

I installed the latest version via [extensions.gnome.org](https://extensions.gnome.org/extension/39/put-windows/), and it gives the dreaded "error."  Below is information about it.  Happy to help troubleshoot.

**Error When Trying to Open Put Windows Settings**
```
SyntaxError: import declarations may only appear at top level of a module @ resource:///org/gnome/Shell/Extensions/js/misc/extensionUtils.js:4:0

Stack trace:
  @file:///home/rsync/.local/share/gnome-shell/extensions/putWindow@clemens.lab21.org/prefs.js:4:24
  _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
```  

** GNOME Information **
```
inxi -Fz                                                                       ✔ 
System:
  Kernel: 6.6.40-1-MANJARO arch: x86_64 bits: 64
  Desktop: GNOME v: 46.3.1 Distro: Manjaro Linux
```

**Comment by negesti** [comment]
Are you using the the correct branch/ the zip file from the pr?
I made some minor adjustments for eslint without restarting the shell but it worked 5 minutes before committing 😕

**Comment by TheRoarkster** [comment]
> Are you using the the correct branch/ the zip file from the pr #209 ? I made some minor adjustments for eslint without restarting the shell but it worked 5 minutes before committing 😕

Let me try again.

**Comment by TheRoarkster** [comment]
You were right; my error.  It works (mostly) with the pr #209.  On `gnome-shell` refresh, the extension is loaded and the settings are available.  I haven't used it for a while, but it seems to be missing the feature where you select where each application is "put."  See attached image of settings.


![1](https://github.com/user-attachments/assets/a8af8ad9-39ee-40ce-979b-a0dcfcdd6ba5)

**Comment by negesti** [comment]
I remove application based config as i personally haven't used if for years, it was dependent on wnck (X11) and caused 90% of the issues.

Fun fact: the extension doesn't work any more today after working without problems on friday and saturday ....

**Comment by ricitron** [comment]
I installed this in Gnome 46 but it doesn't work. I found that when I set the keyboard shortcuts for moving windows they are reset to 'Disabled' immediately when I switch to a different preference tab. So I can't test window movement because there are no keyboard shortcuts.

**Comment by ricitron** [comment]
I installed this in Gnome 46 but it isn't working. When I try to set Keyboard Shortcuts once I close the preferences window and reopen preferences the shortcuts are all reset to 'Disabled'. So I can't test moving windows because there are not shortcuts to do so.

**Comment by negesti** [comment]
Looks like I killed it again during jslint cleanup. 

Or it currently only works using X11 because the new key binding methods are weird

**Comment by ofirnk** [comment]
> I remove application based config as i personally haven't used if for years, it was dependent on wnck (X11) and caused 90% of the issues.
> 
> Fun fact: the extension doesn't work any more today after working without problems on friday and saturday ....

Thanks for your work @negesti :pray: 

I've cloned the repo, and as others have pointed out - it doesn't persist the UI changes to settings.
I'm mostly interested in keyboard shortcuts to move current window to right/left screen, and to center/maximize.

Could you please post an example config and the location to drop it?
Thanks again

**Comment by negesti** [comment]
ESLint auto cleanup "fixed"  ```ret == '1'``` which is obviously fales if ```ret === 1```. Keybinding can be saved now and work on X11 and Wayland. 
Not 100% sure if wayland requires a restart after the settings are stored. Maybe it was only because wayland does not reload the js code after startup.

**Comment by ofirnk** [comment]
Strange - the settings are not persisted for me from the UI (Ubuntu 24 On Wayland). 

> I installed this in Gnome 46 but it isn't working. When I try to set Keyboard Shortcuts once I close the preferences window and reopen preferences the shortcuts are all reset to 'Disabled'. So I can't test moving windows because there are not shortcuts to do so.

@ricitron was it solved for you?

**Comment by negesti** [comment]
Are u using feature/gnom-46?

**Comment by ofirnk** [comment]
Hey @negesti,

I tried switching to feat/gnome-46, but the accelerator settings still weren't sticking—the shortcuts showed as "enabled" but didn't actually work.

I ended up reverting prefs.js to a previous version, and now it's working! :tada: 

Thanks again for everything!

**Comment by ricitron** [comment]
@ofirnk if you got it working can you do a pull request?

**Comment by ricitron** [comment]
> Strange - the settings are not persisted for me from the UI (Ubuntu 24 On Wayland).
> 
> > I installed this in Gnome 46 but it isn't working. When I try to set Keyboard Shortcuts once I close the preferences window and reopen preferences the shortcuts are all reset to 'Disabled'. So I can't test moving windows because there are not shortcuts to do so.
> 
> @ricitron was it solved for you?

It didn't solve it for me yet. 

@ofirnk could you make a pull request with the change that got yours to work?

**Comment by ricitron** [comment]
> @ofirnk @ricitron sry. Setting the keybinding with ` gnome-extensions prefs putWindow@clemens.lab21.org` or using the normal UI should work now

It still does not work for me.

**Comment by negesti** [comment]
That's really strange :(

Can you run ```journalctl -f``` during ``` gnome-extensions prefs putWindow@clemens.lab21.org ``` and check if there is any error beside the annoying gjs warning?

```
gtk_css_node_insert_after: assertion 'previous_sibling == NULL || previous_sibling->parent == parent' failed
gtk_css_node_insert_after: assertion 'previous_sibling == NULL || previous_sibling->parent == parent' failed
g_variant_dup_strv: assertion 'g_variant_is_of_type (value, G_VARIANT_TYPE_STRING_ARRAY)' failed
g_variant_dup_strv: assertion 'g_variant_is_of_type (value, G_VARIANT_TYPE_STRING_ARRAY)' failed
g_variant_dup_strv: assertion 'g_variant_is_of_type (value, G_VARIANT_TYPE_STRING_ARRAY)' failed

```

**Comment by ricitron** [comment]
> That's really strange :(
> 
> Can you run `journalctl -f` during `gnome-extensions prefs putWindow@clemens.lab21.org` and check if there is any error beside the annoying gjs warning?
> 
> ```
> gtk_css_node_insert_after: assertion 'previous_sibling == NULL || previous_sibling->parent == parent' failed
> gtk_css_node_insert_after: assertion 'previous_sibling == NULL || previous_sibling->parent == parent' failed
> g_variant_dup_strv: assertion 'g_variant_is_of_type (value, G_VARIANT_TYPE_STRING_ARRAY)' failed
> g_variant_dup_strv: assertion 'g_variant_is_of_type (value, G_VARIANT_TYPE_STRING_ARRAY)' failed
> g_variant_dup_strv: assertion 'g_variant_is_of_type (value, G_VARIANT_TYPE_STRING_ARRAY)' failed
> ```

I see the following errors:
`gjs[62743]: g_variant_dup_strv: assertion 'g_variant_is_of_type (value, G_VARIANT_TYPE_STRING_ARRAY)' failed`
`gjs[62836]: ../../../gtk/deprecated/gtkliststore.c:863: Unable to convert from gpointer to gint`

**Comment by ricitron** [comment]
OK it works for me now. Thank you.

**Comment by negesti** [comment]
@73 do you stilll use the move focus part of the extension?

**Comment by 73** [comment]
@negesti No, I do not. Thank you so much for asking. That is very considerate of you.
